uqcsbot/minecraft.py

```diff
@@ -93,6 +93,54 @@ async def mcwhitelist(self, interaction: discord.Interaction, username: str):
 
         db_session.close()
 
+    @app_commands.command()
+    @app_commands.describe(username="Minecraft username to unwhitelist.")
+    @yelling_exemptor(input_args=["username"])
+    async def mcunwhitelist(self, interaction: discord.Interaction, username: str):
+        """Removes a username from the whitelist for the UQCS server."""
+        db_session = self.bot.create_db_session()
+        query = db_session.query(MCWhitelist).filter(
+            MCWhitelist.discord_id == interaction.user.id
+        )
+        is_user_admin = (
+            isinstance(interaction.user, Member)
+            and interaction.user.guild_permissions.manage_guild
+        )
+
+        # If the user has already whitelisted someone, and they aren't an admin deny it.
+        if not is_user_admin and query.count() > 0:
```

Comment:
This check is dodgy - it should just be `not is_user_admin`. Otherwise someone who's whitelisted an account can ban anyone they want

Reply:
Subsequently, I guess you could just put `@app_commands.checks.has_permissions(manage_guild=True)` on the whole command and remove the explicit permission check, but that's less of a concern.